test-crawler/extractor/extractor.go

```diff
@@ -203,18 +234,53 @@ func getFunctionNodes(content string, treeCursor *sitter.TreeCursor, parser *a.P
 					continue
 				}
 
-				funcAnnoPair = append(funcAnnoPair, struct {
-					Node *sitter.Node
-					Name string
-				}{
+				funcAnnoPair = append(funcAnnoPair, FunctionAnnotationNode{
 					Node: node,
-					Name: funcName,
+					Function: c.FunctionAnnotation{
+						Name: funcName,
+					},
 				})
 
 				node = nil
 				funcName = ""
 				isIgnored = false
 			}
+			/*
+				Take a look at this function for example:
+				*********
+				func (e *ProofEventStream) ListenProofEvent(
+					ctx context.Context,
+					policy *types2.ProofRegisterPolicy)
+				(<-chan *types2.RequestEvent, error){...}
+				*********
+				First iteration through children nodes:
+				Child.Type = method_declaration
+				Child.Value = whole function body func(*x)(a,b)c{...}
+
+				child.Child(1) is a first part of function declaration, and has type of parameter_list: (e *ProofEventStream)
+				child.Child(2): has type of field_identifier  and represents function's name: ListenProofEvent
+				child.Child(3): has type parameter_list and represents input parameters:  (ctx context.Context, policy *types2.ProofRegisterPolicy)
+				child.Child(4): has type parameter_list and represent return parameters:  (<-chan *types2.RequestEvent, error)
+			*/
+			if child.Type() == string(METHOD_DECLARATION) {
+				funcName = content[child.Child(2).StartByte():child.Child(2).EndByte()]
+				public := unicode.IsUpper(rune(funcName[0]))
```

Comment:
It would be a good idea to extract this to a separate function, e.g. `function isPublic(string functionName)`